src/components/AccessibilityGuide/DynamicContent.stories.mdx

```diff
@@ -0,0 +1,275 @@
+
+**Do not dynamically add or remove the `aria-live` attribute from the element.**
+
+The `aria-live` attribute should be set on the element when the page is rendered.
+The following are **incorrect**:
+
+```html
+<div id="dynamic-content-container">
+  <!-- Any dynamic content should be inserted, removed, or updated here. -->
+</div>
+```
+
+```js
+// Javascript anti-pattern example
+const container = document.getElementById("dynamic-content-container");
+container.setAttribute("aria-live", "polite");
+```
+
+```js
+// React anti-pattern example
+const [ariaLive, setAriaLive] = useState(undefined);
+
+// When the component mounts, set the aria-live attribute.
+useEffect(() => {
+  setAriaLive("polite");
+}, [])
+
+return (
+  <div id="dynamic-content-container" aria-live={ariaLive}>
+    <!-- Any dynamic content should be inserted, removed, or updated here. -->
+  </div>
+);
+```
+
+**Do not assign the `aria-live` attribute to an entire section or the
+entire page.**
+
+Add the `aria-live` attribute to the specific element that will be updated. For
+example, on a search results page that updates the amount of results found and
+the list of results, only add the `aria-live` attribute to the element that
+contains the number of results. The following is not recommended:
+
+```html
+<div id="search-results" aria-live="polite">
+  <div id="results-count">
+    <!-- Dynamic content. -->
+    <span>10 results found</span>
+  </div>
+  <ul id="results-list">
+    <!-- Dynamic list of li elements. -->
+    <li>
+      <p>Title: Result 1</p>
+      <p>Author: Author 1</p>
+    </li>
+    <li>
+      <p>Title: Result 2</p>
+      <p>Author: Author 2</p>
+    </li>
+    <!-- ... -->
+  </ul>
+</div>
+```
+
+This is not recommended because the screen reader will announce:
+
+```
+"10 results found Title: Result 1 Author: Author 1 Title: Result 2 Author: Author 2 ..."
+```
+
+This is information overload. Instead, add the `aria-live` attribute to the
+element that contains the updated number of results found. In this case, it's
+the element with `id` of `results-count` only.
+
+## NYPL Patterns
+
+### HelperErrorText
+
+The `HelperErrorText` component renders an empty `div` element with the
+`aria-live` attribute set to `"polite"` by default. This component should not be
+conditionally rendered on the page. Instead, the content should be dynamically
+updated.
+
+[HelperErrorText Storybook documentation](https://nypl.github.io/nypl-design-system/reservoir/v1/?path=/docs/components-content-display-helpererrortext--helper-error-text-with-controls)
+
+```jsx
+// Very naive example.
+const [text, setText] = useState("");
+
+useEffect(() => {
+  setText("This helper text was dynamically updated.");
+}, []);
+
+// Even though `text` is empty by default, the `HelperErrorText` component
+// is still rendered. The text that is dynamically updated will be announced
+// by assistive technology.
+return <HelperErrorText text={text} />;
+```
+
+### TextInput
```

Comment:
The patterns described here also apply to a number of other form components (i.e. `DatePicker`, `Select`, etc.).  Why did you choose to not include them in the `NYPL Patterns` section?

Reply:
It's the simplest example but this does apply to other form components. I'll update this to "Form Components", list the relevant ones, and still use TextInput as the example.

Reply:
OK.  Yes, including info about the others would be good.